### ISSUE TITLE
[Accessibility]: Screen-reader click on line number doesn't generate canonical link

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36526]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34448

### Problem description

When navigating the code intel blob view, the line numbers are clickable (presumably due to https://github.com/sourcegraph/sourcegraph/issues/35267). However, clicking on them doesn't do anything.

### Expected behavior

I think the right behavior here is to match what would happen if you clicked with a mouse on the line number: in that situation, the line is highlighted and the link becomes canonical.

The screen reader should probably announce the change in the URL for awareness.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by felixfbecker** [comment]
In the CodeMirror blob, the gutter is currently impossible to focus.

**Comment by philipp-spiess** [comment]
Related to: https://github.com/sourcegraph/sourcegraph/issues/43211 We're having a discussion on what we can do about it here: https://github.com/sourcegraph/sourcegraph/pull/44148
